### PR TITLE
refactor(nyt): expose nyt::DB to etl crate

### PR DESCRIPTION
**Pull request by btc**
refactor of nyt::Index/DB. ready for review

### CODE REVIEW

File: pkg/nyt/src/db.rs [hunk 1]
```diff
@@ -1,66 +1,75 @@
...
 
-impl<G, R> Index<G, R>
-where
-    G: Hash + Eq + Copy,
-    R: Hash + Eq + Copy,
-{
-    pub fn new(ingredient_memberships: MapOfSets<G, R>) -> Self {
+impl DB {
+    pub fn load() -> anyhow::Result<Self> {
+        Ok(Self::new(get_ingredients_recipes_map()?))
```
Comment (btc) [inline pkg/nyt/src/db.rs:22]: more refactoring can be done to clean up the layers of initialization we're doing. we can probably flatten this out.

File: pkg/nyt/src/db.rs [hunk 2]
```diff
@@ -1,66 +1,75 @@
...
 
 const DEBUG_ALIASES_PATH: &str = "nyt/not-in-any-recipe.txt";
 
 pub type MapOfSets<A, B> = HashMap<A, HashSet<B>>;
 
-pub struct Index<G, R> {
-    pub recipe_ingredients: MapOfSets<R, G>,
-    pub ingredients_recipes: MapOfSets<G, R>,
+pub struct DB {
+    pub recipe_ingredients: MapOfSets<RecipeID, bible::IngredientID>,
```
Comment (btc) [inline pkg/nyt/src/db.rs:16]: generics have been removed

File: pkg/nyt/src/main.rs [hunk 3]
```diff
@@ -21,7 +21,8 @@ fn run() -> anyhow::Result<()> {
     let ingredients_recipes =
         nyt_searcher.match_ingredients_to_recipes(&ingredient_aliases);
     flavor_common::debug_write_json(&ingredients_recipes, "nyt.json")?;
-    let _index = Index::new(ingredients_recipes);
+    let _index = DB::new(ingredients_recipes);
```
Comment (btc) [inline pkg/nyt/src/main.rs:24]: `nyt/main.rs` can be simplified probably. we can figure out what information it needs and obtain it from the DB.

File: pkg/nyt/src/db.rs [hunk 4]
```diff
@@ -1,66 +1,75 @@
...
-    R: Hash + Eq + Copy,
-{
-    pub fn new(ingredient_memberships: MapOfSets<G, R>) -> Self {
+impl DB {
+    pub fn load() -> anyhow::Result<Self> {
+        Ok(Self::new(get_ingredients_recipes_map()?))
+    }
+
+    pub fn new(
+        ingredient_memberships: MapOfSets<bible::IngredientID, RecipeID>,
```
Comment (ameliariely) [inline pkg/nyt/src/db.rs:26]: Ooo love this name

**Review by ameliariely — APPROVED** [review]

**Pull request merged**